docs/hooks/patient-view.md

```diff
@@ -12,14 +12,18 @@ The user has just opened a patient's record.
 
 ## Context
 
-The patient whose record was opened, including their encounter, if applicable.
+The patient whose record was opened, including their encounter, if applicable. 
+
+The FHIR ids referenced in `context` are expected to be available via the FHIR Server (if `fhirServer` is populated in the request) or via prefetch, if supported. However, the specification does not explicitly require this behavior and implementations should be aware that integrations may not fully support this expectation.
```

Comment:
Suggest:
If the CDS-Client supports pretech and/or provides a FHIR Server, the CDS-Client SHOULD provide FHIR Resources referenced by FHIR ids in `context`. As this behavior is optional, CDS-Server implementers need to be aware that not all CDS-Clients will support this behavior.

Reply:
Is it appropriate to add a statement to the base spec about the nature of context?  e.g. if ids are provided, they must be resolvable through some mechanism available to the CDS Service.  The point being that if an id is present, in order to be useful, the CDS Service must be able to call the FHIR server to retrieve it, or to have that same data prefetched by the CDS Client.

Reply:
Hey Bas, does the rewrite address your concerns? I modified your suggestion for readability and consistency in terminology.

Reply:
First, a few relevant sections of the spec:

* On [CDS Hooks Anatomy](https://cds-hooks.hl7.org/1.0/#cds-hooks-anatomy), a `hook` is 
  > A defined point within the client system's workflow with well-known contextual information provided as part of the request

* On the [HTTP Request](https://cds-hooks.hl7.org/1.0/#http-request_1) to a CDS Service, `context` is 
  > Hook-specific contextual data that the CDS service will need.

* On [Providing FHIR Resources to a CDS Service](https://cds-hooks.hl7.org/1.0/#providing-fhir-resources-to-a-cds-service)
  > Each CDS Service will require specific FHIR resources in order to compute the recommendations the CDS Client requests...this specification defines a process to allow a CDS Service to request and obtain FHIR resources efficiently.  Two optional methods are provided....Each CDS Client SHOULD decide which approach, or combination, is preferred.

* On [Prefetch Template](https://cds-hooks.hl7.org/1.0/#prefetch-template)
  > If the CDS Service is unable to obtain required data because it cannot access the FHIR server and the request did not contain the necessary prefetch keys, the service SHALL respond with an HTTP 412 Precondition Failed status code.

In summary, the spec emphasizes that:
1. A CDS Client's purpose in invoking a hook is to communicate well-known and needed information to the CDS Service
2. CDS Clients SHOULD decide whether to support exposing their FHIR server or implementing prefetch or both.
3. CDS Services SHALL return a 412 if they do not get the information that they need from either the FHIR Server or prefetch (as opposed to returning something unsafe like an empty array of cards).

It isn't the responsibility of a _hook definition_ to warn CDS Services about CDS Clients that choose to limit decision support by preventing access to needed data (or otherwise). Such CDS Clients will not gain adoption and I don't think statements on a hook definition to address their unusable nature add much value.

If we want to add clarifications to a future version of the spec to comment on the nature of FHIR references in a given `context`, that or other similar updates may be helpful.  In the meantime, I don't think the 1.0 specification precludes the ability to publish the existing patient-view definition (prior to these amendments).

@bvdh, thoughts?